modules/nextflow/src/main/groovy/nextflow/script/FunctionDef.groovy

```diff
@@ -33,37 +35,49 @@ class FunctionDef extends ComponentDef implements ChainableDef {
 
     private BaseScript owner
 
-    private Method method
-
     private String name
 
-    FunctionDef(BaseScript owner, Method method) {
+    private String alias
+
+    FunctionDef(BaseScript owner, String name) {
         this.owner = owner
-        this.method = method
-        this.name = method.name
+        this.name = name
+        this.alias = name
     }
 
     protected FunctionDef() { }
 
     String getType() { 'function' }
 
-    Method getMethod() { method }
-
-    String getName() { name }
+    String getName() { alias }
 
     BaseScript getOwner() { owner }
 
     Object invoke_a(Object[] args) {
-        method.invoke(owner, spreadToArray(args))
+        throw new MethodNotSupportedException()
```

Comment:
Not sure about this. What's the rationale?

Reply:
`invoke_a` is an abstract method from ComponentDef and new FunctionDef not use it as it override `invoke_o`

Reply:
But it can via [ChainableDef](https://github.com/nextflow-io/nextflow/blob/20cfbfd3595270b0a71a32fe95de52936bbd475b/modules/nextflow/src/main/groovy/nextflow/script/ChainableDef.groovy#L36-L36), [here](https://github.com/nextflow-io/nextflow/blob/20cfbfd3595270b0a71a32fe95de52936bbd475b/modules/nextflow/src/main/groovy/nextflow/extension/ChannelEx.groovy#L181-L181) and [here](https://github.com/nextflow-io/nextflow/blob/20cfbfd3595270b0a71a32fe95de52936bbd475b/modules/nextflow/src/main/groovy/nextflow/extension/ChannelEx.groovy#L203-L203)

Reply:
If I'm not wrong, and reading the comment of the method, left in both cases is a Process, not a Function

Reply:
of course, the comment is wrong 😄. look at this example 

https://github.com/nextflow-io/nextflow/blob/20cfbfd3595270b0a71a32fe95de52936bbd475b/modules/nextflow/src/test/groovy/nextflow/script/ScriptPipesTest.groovy#L276-L282

Reply:
Ok, fixed and added more test